source/board/microbit.c

```diff
@@ -30,3 +30,47 @@ __attribute__((aligned(4)))
 const char daplink_drive_name[11] = "MICROBIT   ";
 __attribute__((aligned(4)))
 const char *const daplink_target_url = "https://www.microbit.co.uk/device?mbedcode=@B@V";
+
+const char *board_id = "";
+const char * const board_id_mb_1_3 = "9900";
+const char * const board_id_mb_1_5 = "9901";
+
+typedef enum {
+    BOARD_VERSION_1_3 = 0,
+    BOARD_VERSION_1_5 = 1,
+} mb_version_t;
+
+// Enables Board Type Pin, reads it and disables it
+// Depends on gpio_init() to have been executed already
+uint8_t read_board_type_pin(void) {
+    uint8_t pin_state = 0;
+    // GPIO mode, Pull enable, pull down, input
+    PIN_BOARD_TYPE_PORT->PCR[PIN_BOARD_TYPE_BIT] = PORT_PCR_MUX(1) | PORT_PCR_PE(1) | PORT_PCR_PS(0);
+    PIN_BOARD_TYPE_GPIO->PDDR &= ~PIN_BOARD_TYPE;
+    // Read pin
```

Comment:
You might need to add a small delay here after you configure the internal pull-down, before you read the pin. The internal pull-downs are fairly weak at 20-50 kΩ.